lib/puppet/network/http/handler.rb

```diff
@@ -68,7 +68,8 @@ def process(request, response)
     new_response.respond_with(e.status, "application/json", e.to_json)
   rescue StandardError => e
     http_e = Puppet::Network::HTTP::Error::HTTPServerError.new(e)
-    Puppet.err(http_e.message)
+    log_msg = [http_e.message, *http_e.backtrace].join("\n")
+    Puppet.err(log_msg)
```

Comment:
`rescue Exception` isn't encouraged by the Ruby maintainers. Kinda like catching `Throwable` in Java.